pixy/httpapi.go

```diff
@@ -104,16 +106,26 @@ func (as *HTTPAPIServer) handleProduce(w http.ResponseWriter, r *http.Request) {
 	key := getParamBytes(r, routingKey)
 
 	// Get the message body from the HTTP request.
+	if _, ok := r.Header[contentLength]; !ok {
+		replyBadRequest(w, "Missing %s header", contentLength)
+		return
+	}
 	messageSizeStr := r.Header.Get(contentLength)
 	messageSize, err := strconv.Atoi(messageSizeStr)
-	if err != nil || messageSize < 0 {
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte(fmt.Sprintf("Invalid or missing %s header: %s",
-			contentLength, messageSizeStr)))
+	if err != nil {
+		replyBadRequest(w, "Invalid %s header: %s", contentLength, messageSizeStr)
+		return
+	}
+	message, err := readBody(r.Body, int64(100))
```

Comment:
Oups, it was supposed to be messageSize, so we allocate memory only once. But during testing I changed that and forget to return it back.